SignalRServiceBenchmarkPlugin/common/Extensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Common
+{
+    public static class Extensions
+    {
+        public static bool TryGetTypedValue<TKey, TValue, TActual>(
+        this IDictionary<TKey, TValue> data,
+        TKey key,
+        out TActual value, Func<TValue, TActual> converter = null) where TActual : TValue
+        {
+            TValue tmp;
+            if (data.TryGetValue(key, out tmp))
+            {
+                if (converter != null)
+                {
+                    value = converter(tmp);
+                    return true;
+                }
+                if (tmp is TActual)
```

Comment:
What about `{ "a": null }.TryGetTypedValue("a", out string x)`?

Reply:
This extension may have several edge cases, will be covered or narrow down the coverage in next PR.